### PR TITLE
Add support for Eigen Artemis 3D US

### PR DESCRIPTION
**Pull request by fedorov**
Sample dataset will soon be available from a new TCIA collection.

I also have the private dictionary for this manufacturer. Should I create a new EigenArtemis3DUS folder, and add it there? It would be good to keep it around this code for completeness (although it does not rely on those private codes).

### CODE REVIEW

File: DicomUltrasoundPlugin/DicomUltrasoundPlugin.py [hunk 1]
```diff
@@ -327,6 +330,78 @@ def examineGeUSMovie(self, filePath):
...
+    if ds.SOPClassUID != supportedSOPClassUID:
+      # Unsupported class
+      logging.error("sopClassUID "+ds.SOPClassUID+" != supportedSOPClassUID "+supportedSOPClassUID)
+      return []
+
+    if not (ds.Manufacturer == "Eigen" and ds.ManufacturerModelName == "Artemis"):
+      return []
+
+    '''
+    voxelSpacingTag = findPrivateTag(ds, 0x200d, 0x03, "Philips US Imaging DD 036")
```
Comment (lassoan) [inline DicomUltrasoundPlugin/DicomUltrasoundPlugin.py]: Is this correct?

File: DicomUltrasoundPlugin/DicomUltrasoundPlugin.py [hunk 2]
```diff
@@ -327,6 +330,78 @@ def examineGeUSMovie(self, filePath):
...
+    if not voxelSpacingTag:
+      # this is most likely not a PhilipsAffinity image
```
Comment (lassoan) [inline DicomUltrasoundPlugin/DicomUltrasoundPlugin.py]: Probably copy-paste

File: DicomUltrasoundPlugin/DicomUltrasoundPlugin.py [hunk 3]
```diff
@@ -573,6 +650,59 @@ def loadGeUsMovie(self, loadable):
...
+    For X: [-0.5*Rows*PixelAspectRatio[0]/1000, -0.5*Rows*PixelAspectRatio[0]/1000 ]
+
+    For Y: [-0.5*Columns*PixelAspectRatio[0]/1000, -0.5*Columns*PixelAspectRatio[0]/1000 ]
+
+    For Z: [-0.5*NumberOfSlices*PixelAspectRatio[0]/1000, -0.5* NumberOfSlices *PixelAspectRatio[0]/1000 ]
+
+    < end quote
+
+    TODO:
+     * consider checking consistency with the private tags?
```
Comment (lassoan) [inline DicomUltrasoundPlugin/DicomUltrasoundPlugin.py]: What does this mean?

**Comment by lassoan** [comment]
I'll merge this now to avoid merge conflicts (as we have added another module today and we'll reoorganize documentation pages). Please have a look at my comments and follow up with any necessary changes in another pull request.

File: DicomUltrasoundPlugin/DicomUltrasoundPlugin.py (hunk 1 above)
Comment (fedorov) [reply DicomUltrasoundPlugin/DicomUltrasoundPlugin.py]: I left it as a placeholder to add code interpreting some of the private attributes. Private attributes contain information that can also be derived from non-private attributes using "private" process, so I just thought I could check if those two result in the same answer as a consistency check. At the time I was working on this, I did not have an example that includes private attributes, but I do have it now, so I will address this.

File: DicomUltrasoundPlugin/DicomUltrasoundPlugin.py (hunk 2 above)
Comment (fedorov) [reply DicomUltrasoundPlugin/DicomUltrasoundPlugin.py]: as explained above

File: DicomUltrasoundPlugin/DicomUltrasoundPlugin.py (hunk 3 above)
Comment (fedorov) [reply DicomUltrasoundPlugin/DicomUltrasoundPlugin.py]: as explained above

**Comment by fedorov** [comment]
Thank you Andras. So should I make a folder and add that private dictionary file for Eigen Artemis that I have?

**Comment by lassoan** [comment]
If it's a single file and not too big then we could put into https://github.com/SlicerHeart/SlicerHeart/tree/master/DicomUltrasoundPlugin folder. Maybe in the future we will register it with pydicom (and then probably this module would do it).

**Comment by fedorov** [comment]
For the sake of completeness, here's the collection which is now public that contains the Artemis data, which motivated this addition of functionality.

@lassoan I think you might be interested in that collection for the purposes of other projects that were looking for MR-US images. It looks like a very comprehensive dataset.

https://wiki.cancerimagingarchive.net/pages/viewpage.action?pageId=68550661#68550661bcab02c187174a288dbcbf95d26179e8

**Pull request merged**